tensorflow/lite/kernels/conv.cc

```diff
@@ -135,30 +135,78 @@ void TransposeFloatTensor(TfLiteTensor* input, TfLiteTensor* output) {
   }
 }
 
+// Check if im2col needs to be allocated, as some version of optimized Conv dont
+// use it. If any change is supporting im2col in any of the Conv versions, then
+// it should be updated here as well
+bool IsIm2ColRequired(TfLiteTensor* input, TfLiteConvParams* params,
+                      TfLiteTensor* filter, OpData* data, bool is_hybrid,
+                      KernelType kernel_type) {
+  bool need_im2col = (params->stride_width != 1 || params->stride_height != 1 ||
+                      params->dilation_width_factor != 1 ||
+                      params->dilation_height_factor != 1 ||
+                      filter->dims->data[2] != 1 || filter->dims->data[1] != 1);
+
+  // Return early as basic requirement is not met
+  if (!need_im2col) return need_im2col;
+
+  bool is_quantized = input->type == kTfLiteUInt8;
+
+  switch (kernel_type) {
+    case kReference:
+      if (is_hybrid) {
+        return true;
+      } else {
+        return false;
+      }
+    case kGenericOptimized:
+    case kCblasOptimized:
+      return true;
+    case kMultithreadOptimized:
+      if (is_hybrid || is_quantized || data->run_multithreaded_kernel) {
```

Comment:
Hmm, doesn't the multithreaded kernel *not* require im2col?

Reply:
@jdduke : You are right, multithreaded kernel don't need it, but other kernels which are under same code path, need it, so i have kept it.